src/pages/components/RepositoryListComponent.js

```diff
@@ -14,26 +22,80 @@ export default function RepositoryListComponent({
   offset,
   setOffset,
   setLimit,
+  arr,
+  setArr,
+  comment,
+  setComment,
   data,
   onSelectManualReview,
   onSelectSuspeciousMark,
   reFetch,
 }) {
   const minDate = data ? data.date.min : undefined;
+  let [checkAll, setCheckAll] = useState(false);
+  let [actionHidden, setActionHidden] = useState(true);
+  let [commentDisabled, setCommentDisabled] = useState(true);
   const lastFetchedAt = data
     ? moment(data.last_fetched_at).utcOffset(660).toLocaleString()
     : undefined;
   data = data ? data.repositories : undefined;
   let utcTimeOffset = new Date().getTimezoneOffset();
   let utc = utcTimeOffset * -2;
+
   const getRemark = (commits) => {
     let message = "review status changed because of ";
     commits.map((commit, index) => {
       message += `(${index + 1}) ${commit.commit}`;
     });
     return message;
   };
+
+  const markId = (id) => {
+    arr.includes(id) ? arr.splice(arr.indexOf(id), 1) : arr.push(id);
+    setArr(arr);
+    setActionHidden(arr.length == 0);
+    setCommentDisabled(arr.length != 1);
+  };
+
+  const markAll = async (check) => {
```

Comment:
Why does markAll need to be a `async` function?

Reply:
removed